### PR TITLE
fix usage of "changed-files" GitHub action

### PR DESCRIPTION
**Pull request by cedricvanrompay-datadog**
See https://securitylab.github.com/advisories/GHSL-2023-271_changed-files/

### What does this PR do? What is the motivation?

Fix a security vulnerabilit related to the usage of the "changed-files" GitHub action in this repository

### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

**Review by davidejones — APPROVED** [review]

File: .github/workflows/preview_link.yml [hunk 1]
```diff
@@ -29,12 +29,15 @@ jobs:
     - name: Generate links
       env:
         branch: ${{ github.head_ref }}
+        RENAMED_FILES: ${{ steps.changed_files.outputs.renamed_files }}
+        MODIFIED_FILES: ${{ steps.changed_files.outputs.modified_files }}
+        ADDED_FILES: ${{ steps.changed_files.outputs.added_files }}
       id: comment_body
       run: |
         python local/bin/py/preview_links.py --deleted='${{ steps.changed_files.outputs.deleted_files }}' \
```
Comment (hestonhoffman) [inline .github/workflows/preview_link.yml:37]: We don't need it for deleted files here?

Comment (cedricvanrompay-datadog) [reply .github/workflows/preview_link.yml:37]: Good catch

Comment (cedricvanrompay-datadog) [reply .github/workflows/preview_link.yml]: Done in 13a8b35

File: .github/workflows/preview_link.yml [hunk 2]
```diff
@@ -29,12 +29,16 @@ jobs:
...
+        RENAMED_FILES: ${{ steps.changed_files.outputs.renamed_files }}
+        MODIFIED_FILES: ${{ steps.changed_files.outputs.modified_files }}
+        ADDED_FILES: ${{ steps.changed_files.outputs.added_files }}
       id: comment_body
       run: |
-        python local/bin/py/preview_links.py --deleted='${{ steps.changed_files.outputs.deleted_files }}' \
-        --renamed='${{ steps.changed_files.outputs.renamed_files }}' \
-        --modified='${{ steps.changed_files.outputs.modified_files }}' \
-        --added='${{ steps.changed_files.outputs.added_files }}'
+        python local/bin/py/preview_links.py --deleted=$DELETED_FILES\
```
Comment (hestonhoffman) [inline .github/workflows/preview_link.yml:38]:
```suggestion
        python local/bin/py/preview_links.py --deleted=$DELETED_FILES \
```

**Review by hestonhoffman — APPROVED** [review]

**Pull request merged**